src/Avalonia.Controls/TextBlock.cs

```diff
@@ -755,6 +758,17 @@ protected override Size ArrangeOverride(Size finalSize)
                             if (drawable is EmbeddedControlRun controlRun
                                 && controlRun.Control is Control control)
                             {
+                                //Add again to prevent clipping
+                                //Fixes: #17194
+                                foreach (var inline in Inlines!)
```

Comment:
Remove this entirely and instead call `VisualChildren.Add(control)`

Reply:
Call only `VisualChildren.Add(control)`